### DISCUSSION TITLE
How to pass `getCustomTransformers` with new 'zero-config' typescript?

### DISCUSSION THREAD

**Discussion by OutThisLife** [storybookjs/storybook#11341] [Help] [answered]
I'm trying to migrate away from this:

```
[
  {
    loader: 'babel-loader',
    options: {
      presets: ['@babel/preset-env', '@babel/preset-react'],
      plugins: ['babel-plugin-styled-components']
    }
  },

  {
    loader: 'ts-loader',
    options: {
      getCustomTransformers: () => ({
        before: [scTransformer]
      })
    }
  }
]
```

To using `checkOptions` or `reactDocgenTypescriptOptions` I guess. 

Any tips?

**Comment by shilman** [comment, marked as answer]
We use `babel-loader` to process typescript--not `ts-loader`--so you if you want to use custom transformers you'll have to disable zero-config TS (which AFAIK isn't possible yet, but we should add) and stick with your manual configuration.

cc @mrmckeb

**Comment by mrmckeb** [comment]
That's correct, thanks @shilman.

Transformers are not an official feature of TypeScript, and they have no plans to support this in the immediate future from what I've seen. Of course, I completely understand why you may want to use them - I have a TypeScript plugin published on NPM and people regularly ask if they can use it at build time.

As @shilman mentions, if you need to use them, you'll need to reimplement the TypeScript support yourself at this stage, sorry.

To turn this off for now, just remove the existing rules for Babel loader.

**Comment by OutThisLife** [comment]
Confirmed, ty.
